### PR TITLE
UCP/REQUEST: improved logs

### PR DESCRIPTION
**Pull request by hoopoepg**
- added name value into ucp_request in debug mode
- updated request allocation function to add name
- updated string representation of request

### CODE REVIEW

File: src/ucp/core/ucp_request.c [hunk 1]
```diff
@@ -60,13 +60,21 @@ ucp_request_str(ucp_request_t *req, ucs_string_buffer_t *strb, int recurse)
...
     } else {
         ucs_string_buffer_appendf(strb, "no debug info ");
     }
 
     ucs_string_buffer_appendf(
-            strb, "%s memory",
+            strb, "memory %s",
             ucs_memory_type_names[ucp_request_get_mem_type(req)]);
+
+#if UCS_ENABLE_ASSERT
```
Comment (yosefe) [inline src/ucp/core/ucp_request.c]: ENABLE_DEBUG_DATA

File: src/ucp/core/ucp_request.h [hunk 2]
```diff
@@ -320,6 +321,9 @@ struct ucp_request {
             ucp_ep_ext_gen_t      *next_ep; /* Next endpoint to flush */
         } flush_worker;
     };
+#if UCS_ENABLE_ASSERT
```
Comment (yosefe) [inline src/ucp/core/ucp_request.h]: ENABLE_DEBUG_DATA

File: src/ucp/proto/proto_am.inl [hunk 3]
```diff
@@ -472,7 +472,7 @@ ucp_proto_ssend_ack_request_alloc(ucp_worker_h worker, uintptr_t ep_ptr)
 {
     ucp_request_t *req;
 
-    req = ucp_request_get(worker);
+    req = ucp_request_get(worker, "ucp_proto_ssend_ack_request_alloc");
```
Comment (yosefe) [inline src/ucp/proto/proto_am.inl]: ssend_ack

File: src/ucp/rma/amo_send.c [hunk 4]
```diff
@@ -140,7 +140,7 @@ ucs_status_ptr_t ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
         goto out;
     }
 
-    req = ucp_request_get(ep->worker);
+    req = ucp_request_get(ep->worker, "ucp_atomic_fetch_nb");
```
Comment (yosefe) [inline src/ucp/rma/amo_send.c]: maybe remove "ucp_" from all names, since it's all upc requests

File: src/ucp/core/ucp_request.c (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_request.c]: > ENABLE_DEBUG_DATA

renamed macro

File: src/ucp/core/ucp_request.h (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_request.h]: > ENABLE_DEBUG_DATA

renamed macro

File: src/ucp/proto/proto_am.inl (hunk 3 above)
Comment (hoopoepg) [reply src/ucp/proto/proto_am.inl]: > ssend_ack

renamed

File: src/ucp/rma/amo_send.c (hunk 4 above)
Comment (hoopoepg) [reply src/ucp/rma/amo_send.c]: > maybe remove "ucp_" from all names, since it's all upc requests

ok, removed "ucp_" prefix

**Review by yosefe — APPROVED** [review]

**Pull request merged**